berkeley-function-call-leaderboard/bfcl/model_handler/actionGemma_handler.py

```diff
@@ -0,0 +1,144 @@
+import json
+
+from bfcl.model_handler.oss_handler import OSSHandler
+from bfcl.model_handler.model_style import ModelStyle
+
+SYSTEM_PROMPT ="""<bos>
+<start_of_turn>system
+You are an AI assistant for function calling. 
+For politically sensitive questions, security and privacy issues, 
+and other non-computer science questions, you will refuse to answer\n
+""".strip()
+
+TASK_INSTRUCTION = """
+You are an expert in composing functions. You are given a question and a set of possible functions. 
+Based on the question, you will need to make one or more function/tool calls to achieve the purpose. 
+If none of the function can be used, point it out and refuse to answer. 
+If the given question lacks the parameters required by the function, also point it out.
+""".strip()
+
+
+FORMAT_INSTRUCTION = """
+The output MUST strictly adhere to the following JSON format, and NO other text MUST be included.
+The example format is as follows. Please make sure the parameter type is correct. If no function call is needed, please make tool_calls an empty list '[]'
+```
+{
+  "tool_calls": [
+    {"name": "func_name1", "arguments": {"argument1": "value1", "argument2": "value2"}},
+    ... (more tool calls as required)
+  ]
+}
+```
+"""
+
+class ActionGemmaHandler(OSSHandler):
+    def __init__(self, model_name, temperature=0.001, top_p=1, max_tokens=512, dtype="bfloat16") -> None:
+        super().__init__(model_name, temperature, top_p, max_tokens, dtype)
+        self.model_style = ModelStyle.OSSMODEL
+
+    def _format_prompt(query, functions, test_category):
+        def convert_to_xlam_tool(tools):
+            '''Convert the Gorilla function call format to xlam format'''
+            if isinstance(tools, dict):
+                xlam_tools = {
+                    "name": tools["name"],
+                    "description": tools["description"],
+                    "parameters": tools["parameters"].get("properties", {})
+                }
+                required = tools["parameters"].get("required", [])
+                for param in required:
+                    xlam_tools["parameters"][param]["required"] = True
+            elif isinstance(tools, list):
+                xlam_tools = []
+                for tool in tools:
+                    xlam_tools.append(convert_to_xlam_tool(tool))
+            else:
+                xlam_tools = tools
+            return xlam_tools
+
+        tools = convert_to_xlam_tool(functions)
+        if isinstance(tools, dict):
+            tools = [tools]
+
+        content = f"\n{TASK_INSTRUCTION}\n<end_of_turn>\n"
+#         content += f"{FORMAT_INSTRUCTION}\n<end_of_turn>\n\n"
+        content += "<unused0>\n" + json.dumps(tools) + "\n<unused1>\n\n"
+        
+        content += f"<start_of_turn>user\n{query}<end_of_turn>\n\n"
+        return SYSTEM_PROMPT + f"\n{content}\n<start_of_turn>assistant"
```

Comment:
From the model card:
> sample response from applied chat template
> ```
> <bos>
>       <start_of_turn>system
> You are an expert in composing functions. You are given a question and a set of possible functions. 
> Based on the question, you will need to make one or more function/tool calls to achieve the purpose. 
> If none of the functions can be used, point it out and refuse to answer. 
> If the given question lacks the parameters required by the function, also point it out.<end_of_turn>
> 
>       <start_of_turn>user
> अमेरिका के राष्ट्रपति कौन है?<end_of_turn>
> 
>         <unused0>
> [{"name": "get_weather", "description": "Get the current weather for a location", "parameters": {"location": {"type": "string", "description": "The city and state, e.g. San Francisco, New York"}, "unit": {"type": "string", "enum": ["celsius", "fahrenheit"], "description": "The unit of temperature to return"}}},  {"name": "search", "description": "Search for information on the internet", "parameters": {"query": {"type": > "string", "description": "The search query, e.g. 'latest news on AI'"}}}]<unused1>
> 
>   <start_of_turn>assistant
> ```

This seems different from the one formatted here, especially the `SYSTEM_PROMPT`.